client/src/providers/HealthProvider.tsx

```diff
@@ -250,20 +254,45 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
       const provider = new ethers.providers.Web3Provider(connection);
       const signer = provider.getSigner();
       const contract = await fetchContract(signer);
-      const requests = await contract.getRequests();
+
+      const requestsStruct = await contract.getRequests();
+      const requests = requestsStruct.map((request) => ({
+        doctor: request[0],
+        doctorName: request[1],
+        fileName: request[2],
+        status: request[3],
+      }));
       setVerificationRequests(requests);
     } catch (error) {
       setError('Error Loading Health Contract');
     }
   };
 
+  const fetchHealthCareProviders = async () => {
+    try {
+      const web3modal = new Web3Modal();
+      const connection = await web3modal.connect();
+      const provider = new ethers.providers.Web3Provider(connection);
+      const signer = provider.getSigner();
+      const contract = await fetchContract(signer);
+      const hcProviders = await contract.getAllProviders();
+      setProviders(hcProviders);
+    } catch (error) {
+      setError('Error Loading Health Contract');
```

Comment:
please do not set error inside hooks. please throw the error to calling component.